c10/core/TensorImpl.h

```diff
@@ -376,7 +384,7 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
     // caching device on TensorImpl (#12934) to speed up device() calls in all cases.
     const auto tid = type_id();
     if (tid == CPUTensorId() || tid == CUDATensorId() || tid == HIPTensorId() || tid == MSNPUTensorId() ||
-        tid == XLATensorId()) {
+        tid == XLATensorId() || tid == AffineCPUTensorId() || tid == PerChannelAffineCPUTensorId()) {
```

Comment:
just a note that this will conflict with https://github.com/pytorch/pytorch/pull/18578.